packages/cli/src/ui/hooks/useAutoAcceptIndicator.ts

```diff
@@ -43,6 +55,46 @@ export function useAutoAcceptIndicator({
         config.setApprovalMode(nextApprovalMode);
         // Update local state immediately for responsiveness
         setShowAutoAcceptIndicator(nextApprovalMode);
+
+        // Auto-approve pending tool calls when switching to auto-approval modes
```

Comment:
this logic seems like it should be in a more central place than useAutoAcceptIndicator. Can you centralize this with the logic you added on the previous pull request. seems like similar concepts. we need to let useGeminiStream know to handle its queue of things that should be approved but weren't because we just switched settings.